curio/local.py

```diff
@@ -60,15 +60,18 @@
 # The thread-local storage slot that points to the task-local storage dict for
 # whatever task is currently running.
 _current_task_local_storage = threading.local()
-_current_task_local_storage.value = None
 
 
 @contextmanager
 def _enable_tasklocal_for(task):
     # Using a full save/restore pattern here is a little paranoid, but
     # safe. Even if someone does something silly like calling curio.run() from
     # inside a curio coroutine.
-    old = _current_task_local_storage.value
+    try:
+        old = _current_task_local_storage.value
+    except AttributeError:
+        old = _current_task_local_storage.value = None
```

Comment:
doh, this is a silly bug indeed, good catch :-). You could make the patch slightly simpler even by leaving out this assignment to `.value`.